src/main/java/com/bookstore/repository/BookRepository.java

```diff
@@ -1,7 +1,12 @@
 package com.bookstore.repository;
 
 import com.bookstore.entity.Book;
+import java.util.List;
+import java.util.Optional;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface BookRepository extends JpaRepository<Book, Long> {
+    List<Book> findAllByIsDeletedFalse();
```

Comment:
Please fix the soft deleted realization. You should add annotations to your entity class 
```
@SQLDelete(sql = "UPDATE entity SET is_deleted = true WHERE id = ?")
@Where(clause = "is_deleted = false")
```
After you can remove methods from the repository and use Crud repository methods in service directly